include/seqan/bed_io/read_bed.h

```diff
@@ -189,74 +189,93 @@ _readBedRecordNoData(BedRecord<Bed12> & record,
+        clear(buffer);
+        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertNoNewline>());
+        record.itemRgb.green = lexicalCast<int32_t>(buffer);
+        skipOne(iter);
 
-    clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
-    record.itemRgb.blue = lexicalCast<int32_t>(buffer);
-    skipOne(iter);
+        clear(buffer);
+        readUntil(buffer, iter, OrFunctor<IsTab, AssertNoNewline>());
+        record.itemRgb.blue = lexicalCast<int32_t>(buffer);
+        skipOne(iter);
+    }
+    else if (value(iter) == '\t' && record.itemRgb.red == 0)
+    {
+        skipOne(iter);
+        record.itemRgb.green = 0;
+        record.itemRgb.blue = 0;
+    }
+    else
+    {
+        throw ParseError("While parsing field 9 (itemRGB) in BED12 file: "
+                         "value must be either 0 or a list of three integer values r,g,b.");
+    }
 
     // Read BLOCK COUNT
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+    readUntil(buffer, iter, OrFunctor<IsTab, AssertNoNewline>());
     record.blockCount = lexicalCast<int32_t>(buffer);
     skipOne(iter);
 
     // READ BLOCK SIZES
     for (int i = 0; i < record.blockCount - 1; ++i)
     {
         clear(buffer);
-        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertNoNewline>());
         appendValue(record.blockSizes, lexicalCast<int>(buffer));
         skipOne(iter);
     }
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+    readUntil(buffer, iter, OrFunctor<IsTab, OrFunctor<EqualsChar<','>, AssertNoNewline> >());
     appendValue(record.blockSizes, lexicalCast<int>(buffer));
+    if (value(iter) == ',')
+        skipOne(iter);
+
     skipOne(iter);
 
     // READ BLOCK STARTS
     for (int i = 0; i < record.blockCount - 1; ++i)
     {
         clear(buffer);
-        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertFunctor<NotFunctor<IsNewline>, ParseError, Bed> >());
+        readUntil(buffer, iter, OrFunctor<EqualsChar<','>, AssertNoNewline>());
         appendValue(record.blockBegins, lexicalCast<int>(buffer));
         skipOne(iter);
     }
     clear(buffer);
-    readUntil(buffer, iter, OrFunctor<IsTab, IsNewline>());
+    readUntil(buffer, iter, OrFunctor<IsTab, OrFunctor<EqualsChar<','>, IsNewline> >());
     appendValue(record.blockBegins, lexicalCast<int>(buffer));
-    if (isNewline(value(iter)))
-    {
-        skipLine(iter);
-        return;
-    }
-    skipOne(iter);
+    if (!atEnd(iter) && value(iter) == ',')
+        skipOne(iter);
+
+    if (!atEnd(iter) && value(iter) == '\t')
+        skipOne(iter);
+
+    // DO NOT skip line here, because the rest of the line is written into record.data!
```

Comment:
Yes, but it will be skipped at a later time. The `readRecord` function contains
```cpp
readLine(record.data, iter);
```
(line 293), which stores the text behind the 12th column and skips the newline.
If we skipped the newline here, the whole next line would go to `record.data`, which is the erroneous behaviour I want to fix with this PR.

Reply:
OK please add this information in the comment
```suggestion
    // DO NOT skip line here, because the rest of the line (empty or not) is written
    // into record.data within the readRecord function below L:293
```